services/tasks/blockio_linux.go

```diff
@@ -28,11 +28,26 @@ import (
 )
 
 var blockIOEnabled bool
+var blockIOConfigFilePath string
+var blockIOReconfigure bool
 
 func BlockIOEnabled() bool { return blockIOEnabled }
 
-func initBlockIO(configFilePath string) error {
+func BlockIOReconfigure() error {
+	if !blockIOEnabled || !blockIOReconfigure || blockIOConfigFilePath == "" {
+		return nil
+	}
+	err := blockio.SetConfigFromFile(blockIOConfigFilePath, true)
```

Comment:
Thanks @gjkim42, this is definitely a valid point. goresctrl can, for instance, skip scanning io schedulers of those block devices that it already knows.

Another thing that crossed my mind is that it would be nice to re-read block devices only once for a pod. However, it's still possible that something more hackish happens, like an init container creates new block device to the system. This patch is very conservative in this sense: it makes sure that a container always gets the most recent configuration for all block devices when created. I'd leave more complex logic and optimizations to later changes.